src/main/java/uk/gov/hmcts/darts/event/service/impl/CleanupCurrentFlagEventProcessorImpl.java

```diff
@@ -52,6 +68,13 @@ private void supersedeOldEvents(EventRepository.EventIdAndHearingIds eventIdAndH
         if (CollectionUtils.isNotEmpty(eventsToBeSuperseded)) {
             List<Integer> eveIdsThatHaveBeenSuperseded = new ArrayList<>();
             for (EventEntity eventToBeSuperseded : eventsToBeSuperseded) {
+                if (eventToBeSuperseded.getCreatedDateTime().isBefore(earliestIsCurrentClearUpDate)) {
+                    log.info("Event version with event id {} received into modernised DARTS is for a legacy event. Skipping event is_current cleanup " +
+                                 "for eve_id {}",
+                         eventIdAndHearingIds.getEventId(),
+                         eventIdAndHearingIds.getEveId());
+                    continue;
```

Comment:
I think we also need to exclude these from the SQL calls else we always pull back the same data and skip it in code

Reply:
@jackmaloney as discussed, I think there's no point trying to put that logic into the query for the tiny amount of times this is going to run, as we get the benefit of being able to log doing it this way. Happy with this how it is 👍

Reply:
Looking at this again as this only runs for event_ids that we get into the system and not in bulk I am also happy with the above :) as after the first few days of go live we should not get any events that hit legacy data